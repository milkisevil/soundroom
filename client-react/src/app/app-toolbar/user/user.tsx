import * as React from 'react';
import { User as UserModel } from '../../shared/user/user';
import styled from 'styled-components';
import colors from '../../shared/colors/colors';

const User = ({user}: Props) => (
  <UserStyled className="user">
    <img className="o-avatar" src={user.avatar}/> {user.name}
  </UserStyled>
);

export interface Props {
  user: UserModel;
}

const UserStyled = styled.div`
  color: ${colors.greyGrit};
  font-size: 13px;
`;

export default User;
