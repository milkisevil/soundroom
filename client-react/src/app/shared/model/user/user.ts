export interface User {
	_id: string;
	name: string;
	avatar: string;
	created: string;
	modified: string;
	googleId: string;
	spotifyId: string;
	facebookId: string;
	twitterId: string;
	userId: string;
}
