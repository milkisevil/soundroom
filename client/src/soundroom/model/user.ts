export class User {
  
  _id:string;
  name:string;
  avatar:string;
  created:Date;
  modified:Date;
  googleId:string;
  
}
