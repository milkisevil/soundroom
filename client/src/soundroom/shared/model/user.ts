export class User {

  _id: string;

  name: string;

  avatar: string;

  created: Date;

  modified: Date;

  googleId: string;

  spotifyId: string;

  facebookId: string;

  twitterId: string;

  userId: string;

}
