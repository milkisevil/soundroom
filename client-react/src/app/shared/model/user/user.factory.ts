import { User } from './user';

export function createUserFromApiResponse(apiData: any): User {
	const { _id, name, avatar } = apiData;

	return {
		_id,
		name,
		avatar,
		created: apiData.created,
		modified: apiData.modified,
		googleId: apiData.hasOwnProperty('googleId') ? apiData.googleId : null,
		spotifyId: apiData.hasOwnProperty('spotifyId') ? apiData.spotifyId : null,
		facebookId: apiData.hasOwnProperty('facebookId') ? apiData.facebookId : null,
		twitterId: apiData.hasOwnProperty('twitterId') ? apiData.twitterId : null,
		userId: apiData.hasOwnProperty('userId') ? apiData.userId : null
	};
}
