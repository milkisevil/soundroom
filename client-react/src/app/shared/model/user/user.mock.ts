import { User } from './user';

export const MOCK_USER: User = {
  _id: 'mock-user-id',
  name: 'michael jackson',
  avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/brad_frost/128.jpg',
  created: 'Thu May 25 2017 14:57:31 GMT+0100 (BST)',
  modified: 'Fri Jun 02 2017 19:09:31 GMT+0100 (BST)',
  googleId: 'mock-google-id',
  spotifyId: 'mock-spotify-id',
  facebookId: 'mock-facebook-id',
  twitterId: 'mock-twitter-id',
  userId: 'mock-user-id',
};
