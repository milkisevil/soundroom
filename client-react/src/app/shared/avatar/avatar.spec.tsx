import * as React from 'react';
import { create } from 'react-test-renderer';
import { MOCK_USER } from '../model/user/user.mock';
import Avatar from './avatar';

describe('<Avatar />', () => {
  it('should render a medium sized avatar', () => {
    const avatar = create(<Avatar size="medium" user={MOCK_USER}/>).toJSON();
    expect(avatar).toMatchSnapshot();
  });

  it('should render a small sized avatar', () => {
    const avatar = create(<Avatar size="small" user={MOCK_USER}/>).toJSON();
    expect(avatar).toMatchSnapshot();
  });
});
