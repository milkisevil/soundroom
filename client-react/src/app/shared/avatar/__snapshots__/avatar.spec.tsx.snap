// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render a medium sized avatar 1`] = `
<img
  alt="michael jackson"
  className="sc-bdVaJa tXwBa"
  size="medium"
  src="https://s3.amazonaws.com/uifaces/faces/twitter/brad_frost/128.jpg"
  title="michael jackson"
/>
`;

exports[`<Avatar /> should render a small sized avatar 1`] = `
<img
  alt="michael jackson"
  className="sc-bdVaJa fkdRbg"
  size="small"
  src="https://s3.amazonaws.com/uifaces/faces/twitter/brad_frost/128.jpg"
  title="michael jackson"
/>
`;
